File: search/linear-search/linear.java
public class linear {
    public static void main(String args[]) {
        
      //Linear search example 1    
      double itemToFind = 5;
      double[] itemList = {6, 8, 12, 5, 12};
      linearSearch(itemToFind, itemList);
      //Function output:
      //
      //Item was found in list at position 3.
      
      //Linear search example 2
      double itemToFind2 = 85;
      double[] itemList2 = {7,91,65,23,76};
      linearSearch(itemToFind2, itemList2);
      //Function output:
      //
      //Item was not found in list.
      
    }
    
    //Function: findItemInList
    //Input: itemToFind (double), itemList (double[])
    //Output: Index of item in list, or -1 if the item is not in the list
    public static int findItemInList(double itemToFind, double[] itemList) {
        //Iterate through list
        for (int i = 0; i < itemList.length; i++) {
          //Check if current item is item we are looking for
          if (itemList[i] == itemToFind) {
            //Return index of item and finish function
            return i;
          }
        }
        //If no item is found, return -1
        return -1;
    }
  
    //Function: linearSearch
    //Input: itemToFind (double), itemList (double[])
    //Output: Sentence stating result of search.
    public static void linearSearch(double itemToFind, double[] itemList) {
        int itemIndex = findItemInList(itemToFind, itemList);
        if (itemIndex == -1) {
            System.out.println("Item was not found in list.");
        } else {
            System.out.println("Item was found in list at position " + Integer.toString(itemIndex) + ".");
        }
    }

  
}

File: search/linear-search/linear.cpp
#include <iostream>

using namespace std;

//Function: findItemInList
//Input: itemToFind (float), itemList (float[]), arrayLength[int]
//Output: Index of item in list, or -1 if the item is not in the list
int findItemInList(float itemToFind, float itemList[], int arrayLength) {
	//Iterate through array
    for (int i = 0; i < arrayLength; i++) {
    	//Check if current item is the item we are looking for
        if (itemList[i] == itemToFind) {
        	//Return index of item and end function
            return i;
        }
    }
    //If item is not found, return -1
    return -1;
}

//Function: linearSearch
//Input: itemToFind (float), itemList (float[]), arrayLength (int)
//Output: Sentence stating result of search.
void linearSearch(float itemToFind, float itemList[], int arrayLength) {
	int itemIndex = findItemInList(itemToFind, itemList, arrayLength);
	if (itemIndex == -1) {
		cout << "Item was not found in list." << endl;
	} else {
		cout << "Item was found in list at position " << itemIndex << "." << endl;
	}
}

int main()
{
   //Linear search example 1
   float itemList[5] = {1.0,76.0,93.0,51.0,42.0};
   float itemToFind = 93.0;
   int arrayLength = 5;
   linearSearch(itemToFind, itemList, arrayLength);
   //Function output:
   //
   //Item was found in list at position 2.

   //Linear search example 2
   float itemList2[5] = {16.0,7.0,239.0,52.0,2.0};
   float itemToFind2 = 64.0;
   int arrayLength2 = 5;
   linearSearch(itemToFind2, itemList2, arrayLength2);
   //Function output:
   //
   //Item was not found in list..
   
   return 0;
}
